lib/irb/cmd/debug.rb

```diff
@@ -5,28 +5,68 @@ module IRB
 
   module ExtendCommand
     class Debug < Nop
+      BINDING_IRB_FRAME_REGEXPS = [
+        '<internal:prelude>',
+        binding.method(:irb).source_location.first,
+      ].map { |file| /\A#{Regexp.escape(file)}:\d+:in `irb'\z/ }
+      IRB_DIR = File.expand_path('..', __dir__)
+
       def execute(*args)
-        require "debug/session"
-        DEBUGGER__.start(nonstop: true)
-        DEBUGGER__.singleton_class.send(:alias_method, :original_capture_frames, :capture_frames)
-
-        def DEBUGGER__.capture_frames(skip_path_prefix)
-          frames = original_capture_frames(skip_path_prefix)
-          frames.reject! do |frame|
-            frame.realpath&.start_with?(::IRB::Irb::DIR_NAME) || frame.path.match?(/internal:prelude/)
-          end
-          frames
+        unless binding_irb?
+          puts "`debug` command is supported only on binding.irb"
```

Comment:
nitpick: I think the message could be a bit more explicit:

```
`debug` command is only available when IRB is started with binding.irb
```

Reply:
```suggestion
          puts "`debug` command is only available when IRB is started with binding.irb"
```